packages/backend/src/accounting/service.ts

```diff
@@ -68,3 +68,166 @@ export interface AccountingService {
   postWithdrawal(id: string): Promise<void | TransferError>
   voidWithdrawal(id: string): Promise<void | TransferError>
 }
+
+export interface TransferToCreate {
+  sourceAccountId: string
+  destinationAccountId: string
+  amount: bigint
+  ledger: number
+}
+
+interface CreateAccountToAccountTransferArgs {
+  transferArgs: TransferOptions
+  voidTransfers(transferIds: string[]): Promise<void | TransferError>
+  postTransfers(transferIds: string[]): Promise<void | TransferError>
+  getAccountReceived(accountRef: string): Promise<bigint | undefined>
+  createPendingTransfers(
+    transfers: TransferToCreate[]
+  ): Promise<string[] | TransferError>
+  withdrawalThrottleDelay?: number
+}
+
+export async function createAccountToAccountTransfer(
+  args: CreateAccountToAccountTransferArgs
+): Promise<Transaction | TransferError> {
+  const {
+    voidTransfers,
+    postTransfers,
+    createPendingTransfers,
+    getAccountReceived,
+    withdrawalThrottleDelay,
+    transferArgs
+  } = args
+
+  const { sourceAccount, destinationAccount, sourceAmount, destinationAmount } =
+    transferArgs
+
+  if (sourceAccount.id === destinationAccount.id) {
+    return TransferError.SameAccounts
+  }
+
+  if (sourceAmount <= 0n) {
+    return TransferError.InvalidSourceAmount
+  }
+
+  if (destinationAmount !== undefined && destinationAmount <= 0n) {
+    return TransferError.InvalidDestinationAmount
+  }
+
+  const transfersToCreateOrError =
+    sourceAccount.asset.ledger === destinationAccount.asset.ledger
+      ? buildSameAssetTransfers(transferArgs)
+      : buildCrossAssetTransfers(transferArgs)
+
+  if (isTransferError(transfersToCreateOrError)) {
+    return transfersToCreateOrError
+  }
+
+  const pendingTransferIdsOrError = await createPendingTransfers(
+    transfersToCreateOrError
+  )
+
+  if (isTransferError(pendingTransferIdsOrError)) {
+    return pendingTransferIdsOrError
+  }
+
+  return {
+    post: async (): Promise<void | TransferError> => {
+      const error = await postTransfers(pendingTransferIdsOrError)
+
+      if (error) {
+        return error
+      }
+
+      if (destinationAccount.onCredit) {
+        const totalReceived = await getAccountReceived(destinationAccount.id)
+
+        if (totalReceived === undefined) {
+          throw new Error()
+        }
+
+        await destinationAccount.onCredit({
+          totalReceived,
+          withdrawalThrottleDelay
+        })
```

Comment:
It does feel out of place, especially in comparison with the `void`.
I'll think of a few ideas, my very first reaction was that this could be some kind of event (via some infrastructure if we were using some pub-sub pattern or just local eventEmitter)

Reply:
https://github.com/interledger/rafiki/issues/241#issuecomment-1337324626
>Or maybe this could be replaced by an issue to use Fynbos's beloved https://temporal.io/